src/Sylius/Bundle/CoreBundle/Form/EventSubscriber/CustomerRegistrationFormSubscriber.php

```diff
@@ -33,29 +32,26 @@ public static function getSubscribedEvents(): array
         ];
     }
 
-    /**
-     * @throws \InvalidArgumentException
-     */
     public function preSubmit(FormEvent $event): void
     {
         $rawData = $event->getData();
         $form = $event->getForm();
         $data = $form->getData();
 
-        Assert::isInstanceOf($data, CustomerInterface::class);
+        if ($data instanceof CustomerInterface) {
+            // if email is not filled in, go on
+            if (!isset($rawData['email']) || empty($rawData['email'])) {
+                return;
+            }
```

Comment:
```suggestion
            if (empty($rawData['email'])) {
                return;
            }
```
Should do the same work.